File: 프로그래머스/unrated/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter
def solution(topping):
    answer = 0
    a = Counter(topping)
    b = {}
    for i in range(len(topping)):
        if topping[i] in a.keys():
            a[topping[i]] -=1
            if a[topping[i]] ==0:
                del a[topping[i]]
            if topping[i] in b.keys():
                b[topping[i]]+=1
            else:
                b[topping[i]]=1
        if len(a)==len(b):
            answer+=1
    return answer

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.js
function solution(topping) {
	const map = new Map();
    const bro = new Set();
    let answer = 0;
    for(let i=0; i<topping.length; i++){
        map.has(topping[i]) ? map.set(topping[i], map.get(topping[i])+1) : map.set(topping[i], 1);
    }
    for(let i=0; i<topping.length; i++){
        let count = map.get(topping[i])-1;
        bro.add(topping[i]);
        count === 0 ? map.delete(topping[i]) : map.set(topping[i], count);
        if(bro.size === map.size) answer++;

    }
    return answer;
}